File: python/problem_242.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        sMap = dict()
        for char in s:
            sMap[char] = sMap.get(char, 0) + 1

        for char in t:
            if char in sMap and sMap[char] > 0:
                sMap[char] = sMap[char] - 1
            else:
                return False
        return True


if __name__ == "__main__":
    s = Solution()
    assert s.isAnagram("anagram", "nagaram"), "Test case 1 failed"
    assert not s.isAnagram("rat", "car"), "Test case 2 failed"
    print("All test cases passed")

File: go/problem_242.go
package main

func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}

	sMap := make(map[rune]int)
	for _, charRune := range s {
		if val, ok := sMap[charRune]; ok {
			sMap[charRune] = val + 1
		} else {
			sMap[charRune] = 1
		}
	}

	for _, charRune := range t {
		if val, ok := sMap[charRune]; ok && val > 0 {
			sMap[charRune] = val - 1
		} else {
			return false
		}
	}

	return true
}
